### PR TITLE
Add support for job time to live

### PR DESCRIPTION
**Pull request by balbirthomas**
Failure to remove resources used by completed jobs/pods can lead to resource exhaustion in the Kubernetes cluster. This commit adds support for setting a ttlSecondsAfterFinished parameter in the Kubernetes Job sepc. This parameter specifies the amount of time (in seconds) for which the resources used by completed job/pod will persist after it which it will be garbage collected by the Kube control manager.

The newly added parameter is optional and only set if it passed in through the `config` argument to the `make_job_spec` method.

## Related links

### CODE REVIEW

File: kubetools/kubernetes/config/job.py [hunk 1]
```diff
@@ -99,3 +99,8 @@ def make_job_config(
             },
         },
     }
+
+    if config.get('ttlSecondsAfterFinished'):
```
Comment (gchazot) [inline kubetools/kubernetes/config/job.py]: `if 'ttlSecondsAfterFinished' in config`:

**Review by gchazot — COMMENTED** [review]
Please check if tests can be added for this new flag, and if any documentation/samples needs updated

File: kubetools/kubernetes/config/job.py (hunk 1 above)
Comment (balbirthomas) [reply kubetools/kubernetes/config/job.py]: Ack. Will change.

**Comment by balbirthomas** [comment]
> Please check if tests can be added for this new flag, and if any documentation/samples needs updated

Ack. Will do and ping you for review.

File: kubetools/kubernetes/config/job.py (hunk 1 above)
Comment (balbirthomas) [reply kubetools/kubernetes/config/job.py]: Changes made and set for review.

**Review by gchazot — APPROVED** [review]
LGTM.
Please be aware that, even when this is released into a new version of kubetools, it will still [not work on clusters running k8s < 1.21](https://github.com/EDITD/kubetools/issues/130).

One solution may be to backport this onto a branch based on `kubetools 13.8.1` and release that as a 13.8.x version

**Review by seangarner — APPROVED** [review]

**Pull request merged**